Add torch.memory_format support to the TorchScript

VitalyFedyunin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29461 [WIP] Experiments
* #29460 [WIP] Switch default of clone, to, *_like ops
* #29459 [WIP] Prepare copy op to switch memory format preservation defaults
* #28991 Adding function to convert Module to channels last
* **#28544 Add torch.memory_format support to the TorchScript**
* #28292 Add memory format support to the `resize_` op.
* #28291 TensroIterator preserve format for binary, ternary operators.
* #28077 Add channels last support to cuda.comm.scatter and gather
* #28076 Kill `operator==` of TensorOptions as confusing one
* #27979 Add memory format support to `resize_as_` operator

Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

VitalyFedyunin committed:
Add torch.memory_format support to the TorchScript

[ghstack-poisoned]

VitalyFedyunin committed:
randn_like mappable again and supported in jit with memory format on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
fix bc test for empty_like on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
fix bc test for empty_like now with gradient on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
rebase to master on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

zdevito reviewed (commented):
It's hard to know if this is correct without tests, but the code looks simple enough.

ezyang commented:
Yeah a test, or even just some basic indication how you figured out this was working (e.g., does a later PR exercise it?) would be great.

VitalyFedyunin committed:
add TensroIterator tests on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
Add some tests on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
Addressed PR feedback on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
Changed model conversion code on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
Linter fix on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
Fix resize_ on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
Fix TensorIterator on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
Fix Autograd on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin commented:
Test added

VitalyFedyunin committed:
Fix resize_as_ with new test on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
Fix resize_as_ on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
Fix resize_ on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
resize_ PR fixes on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
Add resize_as_ docs on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
fix resize_as_ docs on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]

VitalyFedyunin committed:
fix resize_as_ docs on "Add torch.memory_format support to the TorchScript"


Differential Revision: [D18093801](https://our.internmc.facebook.com/intern/diff/D18093801)

[ghstack-poisoned]
